pkg/kubelet/kubelet.go

```diff
@@ -901,6 +901,23 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		// If the cloud has zone information, label the node with the zone information
+		zones, ok := kl.cloud.Zones()
```

Comment:
@justinsb: To make kubelet support for custom labels extensible we need an `exec` based plugin model. This PR can go in as-is, but once we get around to adding support for custom label plugins, this code needs to be refactored.